pkgs/applications/virtualization/colima/default.nix

```diff
@@ -13,31 +17,39 @@ buildGo118Module rec {
   src = fetchFromGitHub {
     owner = "abiosoft";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-66nKH5jxTzLB9bg2lH1E8Cc0GZ6C/N/+yPYhCVEKOBY=";
-
+    rev = "1cfe5f21ed3769f3248945cb886b73ed3f3f816d";
+    sha256 = "kKLqapQbmwmuJ4Vfgjd9HjCVtFIdHpTORI4FSixJGaw=";
     # We need the git revision
     leaveDotGit = true;
-    postFetch = ''
-      git -C $out rev-parse HEAD > $out/.git-revision
-      rm -rf $out/.git
-    '';
   };
 
-  nativeBuildInputs = [ installShellFiles makeWrapper ];
+  nativeBuildInputs = [ installShellFiles makeWrapper git openssl ];
+
+  vendorSha256 = "jDzDwK7qA9lKP8CfkKzfooPDrHuHI4OpiLXmX9vOpOg=";
 
-  vendorSha256 = "sha256-91Ex3RPWxOHyZcR3Bo+bRdDAFw2mEGiC/uNKjdX2kuw=";
+  CGO_ENABLED = 0;
 
-  doCheck = false;
+  buildPhase = ''
+    runHook preBuild
+    make build VERSION=${version} REVISION=$(git rev-parse --short HEAD)
+    runHook postBuild
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+    make test
+    runHook postCheck
+  '';
 
-  preConfigure = ''
-    ldflags="-X github.com/abiosoft/colima/config.appVersion=${version}
-              -X github.com/abiosoft/colima/config.revision=$(cat .git-revision)"
+  installPhase = ''
+    runHook preInstall
+    make install INSTALL_DIR=$out/bin
+    runHook postInstall
   '';
 
   postInstall = ''
     wrapProgram $out/bin/colima \
-      --prefix PATH : ${lib.makeBinPath [ lima ]}
+      --prefix PATH : ${lib.makeBinPath [ lima-unwrapped qemu ]}
```

Comment:
You'd probably still have the two separate derivations, I'm just saying the wrapped derivation could expose some more stuff. I'll try make some changes and give an example

Reply:
ok here we go:

colima using `lima` but still has `qemu`

```
λ cat ./result/bin/colima
#! /nix/store/j9rc786ylq8cid4zgcn6idknilkbd4ax-bash-5.1-p16/bin/bash -e
PATH=${PATH:+':'$PATH':'}
PATH=${PATH/':''/nix/store/sa8sd1isamxybk6f55r2k10f2pmpi5pi-qemu-7.0.0/bin'':'/':'}
PATH='/nix/store/sa8sd1isamxybk6f55r2k10f2pmpi5pi-qemu-7.0.0/bin'$PATH
PATH=${PATH#':'}
PATH=${PATH%':'}
export PATH
PATH=${PATH:+':'$PATH':'}
PATH=${PATH/':''/nix/store/90104vaymsdjy62gvxwdr1pz656pcy79-lima-0.11.1/bin'':'/':'}
PATH='/nix/store/90104vaymsdjy62gvxwdr1pz656pcy79-lima-0.11.1/bin'$PATH
PATH=${PATH#':'}
PATH=${PATH%':'}
export PATH
exec -a "$0" "/nix/store/xav70i18613vg4v4c9lnw7p8a77gm2j1-colima-0.4.2/bin/.colima-wrapped"  "$@"

λ ls -al /nix/store/90104vaymsdjy62gvxwdr1pz656pcy79-lima-0.11.1/bin/
dr-xr-xr-x root root 4.0 KB Thu Jan  1 01:00:01 1970  .
dr-xr-xr-x root root 4.0 KB Thu Jan  1 01:00:01 1970  ..
.r-xr-xr-x root root 561 B  Thu Jan  1 01:00:01 1970  docker.lima
.r-xr-xr-x root root 950 B  Thu Jan  1 01:00:01 1970  lima
.r-xr-xr-x root root  12 MB Thu Jan  1 01:00:01 1970  limactl
.r-xr-xr-x root root  41 B  Thu Jan  1 01:00:01 1970  nerdctl.lima
.r-xr-xr-x root root 576 B  Thu Jan  1 01:00:01 1970  podman.lima
```

rather than `lima-wrapper`


```
λ nix-build -A lima
this derivation will be built:
  /nix/store/ajlgddwjcmn5hlgmn6zdypj8mvwc47p9-lima-wrapper-0.11.1.drv
building '/nix/store/ajlgddwjcmn5hlgmn6zdypj8mvwc47p9-lima-wrapper-0.11.1.drv'...
/nix/store/n637krbwz0y4qdjflvzlx26hxnfyv4w5-lima-wrapper-0.11.1
λ ls -al ./result/bin
dr-xr-xr-x root root 4.0 KB Thu Jan  1 01:00:01 1970  .
dr-xr-xr-x root root 4.0 KB Thu Jan  1 01:00:01 1970  ..
.r-xr-xr-x root root 381 B  Thu Jan  1 01:00:01 1970  limactl
λ ls -al ./result
lrwxrwxrwx jk users 63 B Wed Jun 29 15:28:08 2022  ./result ⇒ /nix/store/n637krbwz0y4qdjflvzlx26hxnfyv4w5-lima-wrapper-0.11.1
```

<details>
  <summary>patch</summary>

```patch
diff --git a/pkgs/applications/virtualization/colima/default.nix b/pkgs/applications/virtualization/colima/default.nix
index 4b756bb720a..a112c02e19d 100644
--- a/pkgs/applications/virtualization/colima/default.nix
+++ b/pkgs/applications/virtualization/colima/default.nix
@@ -1,12 +1,12 @@
 { lib
-, buildGo118Module
+, buildGoModule
 , fetchFromGitHub
 , installShellFiles
 , lima
 , makeWrapper
 }:
 
-buildGo118Module rec {
+buildGoModule rec {
   pname = "colima";
   version = "0.4.2";
 
@@ -23,21 +23,27 @@ buildGo118Module rec {
       rm -rf $out/.git
     '';
   };
+  vendorSha256 = "sha256-91Ex3RPWxOHyZcR3Bo+bRdDAFw2mEGiC/uNKjdX2kuw=";
 
   nativeBuildInputs = [ installShellFiles makeWrapper ];
 
-  vendorSha256 = "sha256-91Ex3RPWxOHyZcR3Bo+bRdDAFw2mEGiC/uNKjdX2kuw=";
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/abiosoft/colima/config.appVersion=${version}"
+  ];
 
-  doCheck = false;
 
-  preConfigure = ''
-    ldflags="-X github.com/abiosoft/colima/config.appVersion=${version}
-              -X github.com/abiosoft/colima/config.revision=$(cat .git-revision)"
+  preBuild = ''
+    ldflags+=" -X github.com/abiosoft/colima/config.revision=$(cat .git-revision)"
   '';
 
+  doCheck = false;
+
+  # colima requires an unwrapped lima but we still want the packages lima was wrapped in to be available too
   postInstall = ''
     wrapProgram $out/bin/colima \
-      --prefix PATH : ${lib.makeBinPath [ lima ]}
+      --prefix PATH : ${lib.makeBinPath ([ lima.passthru.lima-unwrapped ] ++ lima.passthru.wrapperPackages)}
 
     installShellCompletion --cmd colima \
       --bash <($out/bin/colima completion bash) \
diff --git a/pkgs/applications/virtualization/lima/default.nix b/pkgs/applications/virtualization/lima/default.nix
index c6b9b4149fe..c9efdb98fab 100644
--- a/pkgs/applications/virtualization/lima/default.nix
+++ b/pkgs/applications/virtualization/lima/default.nix
@@ -2,8 +2,6 @@
 , buildGoModule
 , fetchFromGitHub
 , installShellFiles
-, qemu
-, makeWrapper
 }:
 
 buildGoModule rec {
@@ -19,7 +17,7 @@ buildGoModule rec {
 
   vendorSha256 = "sha256-wvb592mmxOeRSX8Rxx2m7tZ2S8wQTcQkL3b6z0F6OEQ=";
 
-  nativeBuildInputs = [ makeWrapper installShellFiles ];
+  nativeBuildInputs = [ installShellFiles ];
 
   # clean fails with read only vendor dir
   postPatch = ''
@@ -32,12 +30,14 @@ buildGoModule rec {
     runHook postBuild
   '';
 
+  # functions required for the default checkPhase are added in the buildPhase
+  # which we overwrite
+  doCheck = false;
+
   installPhase = ''
     runHook preInstall
     mkdir -p $out
     cp -r _output/* $out
-    wrapProgram $out/bin/limactl \
-      --prefix PATH : ${lib.makeBinPath [ qemu ]}
     installShellCompletion --cmd limactl \
       --bash <($out/bin/limactl completion bash) \
       --fish <($out/bin/limactl completion fish) \
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index d6f1798ad64..df049c57526 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -34458,9 +34458,13 @@ with pkgs;
 
   idsk = callPackage ../tools/filesystems/idsk { stdenv = gcc10StdenvCompat; };
 
-  colima = callPackage ../applications/virtualization/colima { };
+  colima = callPackage ../applications/virtualization/colima {
+    buildGoModule = buildGo118Module;
+  };
+
+  lima = callPackage ../applications/virtualization/lima/wrapper.nix { };
 
-  lima = callPackage ../applications/virtualization/lima { };
+  lima-unwrapped = callPackage ../applications/virtualization/lima { };
 
   logtop = callPackage ../tools/misc/logtop { };
 
```

</details>

it can be made robust so that it can handle being passed an unwrapped lima like this:

```nix
{ pkgs ? import ./. { } }:

pkgs.colima.override { lima = pkgs.lima-unwrapped; }
```

if changed like this:

```
let
  isLimaWrapped = builtins.hasAttr "wrapperPackages" lima.passthru;
  lima-unwrapped = if isLimaWrapped then lima.passthru.lima-unwrapped else lima;
  # colima requires an unwrapped lima but we still want the packages lima was
  # wrapped in (if it was a wrapped lima) to be available too
  packages = [
    lima-unwrapped
  ] ++ lib.optionals (isLimaWrapped) lima.passthru.wrapperPackages;
in
```

&&

```
postInstall = ''
    wrapProgram $out/bin/colima \
      --prefix PATH : ${lib.makeBinPath packages}
# ...
```

I'll push the fully robust one to a branch and link it since that might be easier to scroll through than the patch file output

Reply:
@06kellyjac This is not needed anymore since we fixed it upstream😃.